### ISSUE TITLE
MPS SDPA returns NaN when attention mask blocks all rows

### ISSUE THREAD

**Issue by manueldeprada** [pytorch/pytorch#156707]
### 🐛 Describe the bug

When using scaled_dot_product_attention on MPS with attention masks that block all attention for a position, the function returns NaNs instead of zeros. This creates inconsistent behavior between CPU and MPS backends.

## Root Cause
In `aten/src/ATen/native/mps/operations/Attention.mm`, the `sdpa_general_mps` function directly calls Metal's `softMaxWithTensor` without handling the edge case where all attention scores are `-inf`. This occurs when an attention mask has `False` for all positions that a query can attend to.

I think the issue is here, when passing all-`-inf` tensor to Metal softmax, which produces NaNs
https://github.com/pytorch/pytorch/blob/e2c9d8d6414927ce754bbc40b767edf103cf16da/aten/src/ATen/native/mps/operations/Attention.mm#L117

While CPU code explicitly checks for `tmp_max == -infinity` and returns zeros:
https://github.com/pytorch/pytorch/blob/e2c9d8d6414927ce754bbc40b767edf103cf16da/aten/src/ATen/native/cpu/FlashAttentionKernel.cpp#L649-L653

## Test Case
```python
import torch
import torch.nn.functional as F

# Minimal reproduction
q = torch.randn(1, 1, 2, 4)
k = torch.randn(1, 1, 2, 4)
v = torch.randn(1, 1, 2, 4)
mask = torch.tensor([[[[False, False], [True, True]]]], dtype=torch.bool)

# CPU returns zeros
out_cpu = F.scaled_dot_product_attention(q, k, v, attn_mask=mask)
print("CPU:", out_cpu[0, 0, 0, :3].tolist())  # [0.0, 0.0, 0.0]

# MPS returns NaNs
out_mps = F.scaled_dot_product_attention(q.to('mps'), k.to('mps'), v.to('mps'), attn_mask=mask.to('mps'))
print("MPS:", out_mps[0, 0, 0, :3].tolist())  # [nan, nan, nan]
```

## Impact
This kind of situation happens when you have left-padded batches for generation in HF's transformers.

tagging the mps people: @jhavukainen @kulinseth @malfet

### Versions

PyTorch version: 2.7.1
Is debug build: False
CUDA used to build PyTorch: None
ROCM used to build PyTorch: N/A

OS: macOS 15.5 (arm64)
GCC version: Could not collect
Clang version: 17.0.0 (clang-1700.0.13.5)
CMake version: version 4.0.2
Libc version: N/A

Python version: 3.13.5 | packaged by conda-forge | (main, Jun 16 2025, 08:24:05) [Clang 18.1.8 ] (64-bit runtime)
Python platform: macOS-15.5-arm64-arm-64bit-Mach-O
Is CUDA available: False
CUDA runtime version: No CUDA
CUDA_MODULE_LOADING set to: N/A
GPU models and configuration: No CUDA
Nvidia driver version: No CUDA
cuDNN version: No CUDA
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

CPU:
Apple M3

Versions of relevant libraries:
[pip3] numpy==2.3.1
[pip3] torch==2.7.1

cc @kulinseth @albanD @malfet @DenisVieriu97 @jhavukainen

**Comment by jhavukainen** [comment]
I think you are correct with the reasoning. We can compute a mask for expected NaNs from the softmax (if all entries along axis are inf) and use it to overwrite the values with zeros before the final matmul. Let me see about a small workaround for the case.

**Comment by jhavukainen** [comment]
Added a workaround and included your repro code to the test cases here: https://github.com/pytorch/pytorch/pull/157727
